websites/management/commands/fix_file_urls.py

```diff
@@ -0,0 +1,127 @@
+        )
+        parser.add_argument(
+            "-p",
+            "--prefix",
+            dest="prefix",
+            default=None,
+            help="Prefix string for WebsiteContent query",
+            required=True,
+        )
+        parser.add_argument(
+            "-ss",
+            "--skip-sync",
+            dest="skip_sync",
+            action="store_true",
+            default=False,
+            help="Whether to skip running the sync_unsynced_websites task",
+        )
+
+    def handle(self, *args, **options):
+        commit_changes = options["commit"]
+        prefix = options["prefix"]
+        csv_output = options["out"]
+
+        modified_content = []
+        bad_paths = WebsiteContent.objects.filter(
+            type=CONTENT_TYPE_RESOURCE,
+            file__regex=r"^/?{}/[A-Za-z0-9\-\.\_]+(\.).*".format(prefix),
+        )
+
+        self.stdout.write(
+            f"Found {bad_paths.count()} resources with '{prefix}/' file paths missing website names"
+        )
+
+        s3_bucket = boto3.resource("s3").Bucket(name=settings.AWS_STORAGE_BUCKET_NAME)
+        for content in bad_paths:
+            new_path = re.sub(
+                r"^(/?{}/)(.*)".format(prefix),
+                r"{}/{}/\2".format(prefix, content.website.name),
+                content.file.name,
+            )
+            file_exists = len(list(s3_bucket.objects.filter(Prefix=new_path))) == 1
+            content_summary = {
+                "website": content.website.name,
+                "content": content.text_id,
+                "original_path": content.file,
+                "new_path": new_path,
+                "exists": file_exists,
+            }
+            if not file_exists:
+                self.stderr.write(f"Content not found at new path: {content_summary}.")
+            modified_content.append(content_summary)
+            if commit_changes and file_exists:
+                content.file = new_path
+                content.save()
+
+        if csv_output and modified_content:
+            self.stdout.write(f"Writing affected content to csv file {csv_output}")
+            self.write_to_csv(csv_output, modified_content)
+
+        if (
+            settings.CONTENT_SYNC_BACKEND
+            and commit_changes
+            and modified_content
+            and not options["skip_sync"]
+        ):
+            self.stdout.write("Syncing all unsynced content to the designated backend")
+            start = now_in_utc()
+            task = sync_unsynced_websites.delay(create_backends=True)
+            self.stdout.write(f"Starting task {task}...")
+            task.get()
+            total_seconds = (now_in_utc() - start).total_seconds()
+            self.stdout.write(
+                "Backend sync finished, took {} seconds".format(total_seconds)
+            )
+
+        self.stdout.write(
+            f"Done updating {bad_paths.count()} resources, with commit={commit_changes}"
```

Comment:
You probably don't want this output if you're just writing to csv

Reply:
Changing to "Finished with commit={commit_changes}"